### ISSUE TITLE
Bundler/DuplicatedGroup undefined method

### ISSUE THREAD

**Issue by Carlgo11** [rubocop/rubocop#12111]
## Expected behavior

Preferably, to Rubocop not to return errors. 

```SH
$ bundle exec rubocop
Inspecting 15 files
...............

15 files inspected, no offenses detected
$ echo $?
0
```

## Actual behavior

```SH
$ bundle exec rubocop
An error occurred while Bundler/DuplicatedGroup cop was inspecting /home/user/Git/2factorauth/twofactorauth/Gemfile.
To see the complete backtrace run rubocop -d.
Inspecting 15 files
An error occurred while Bundler/DuplicatedGroup cop was inspecting /home/user/Git/2factorauth/twofactorauth/Gemfile.
To see the complete backtrace run rubocop -d.
$ echo $?
1
...............

15 files inspected, no offenses detected

1 error occurred:
An error occurred while Bundler/DuplicatedGroup cop was inspecting /home/user/Git/2factorauth/twofactorauth/Gemfile.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.56.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.2) [x86_64-linux-gnu]
```

```
Scanning /home/user/Git/2factorauth/twofactorauth/Gemfile
An error occurred while Bundler/DuplicatedGroup cop was inspecting /home/user/Git/2factorauth/twofactorauth/Gemfile.
undefined method `value' for s(:hash,
  s(:pair,
    s(:sym, :optional),
    s(:true))):RuboCop::AST::HashNode

            .group_by { |node| node.arguments.map(&:value).map(&:to_s).sort }
                                             ^^^^
Did you mean?  values
```
## Steps to reproduce the problem

Run Rubocop (using a version where #12074 is included) with the following Gemfile:
```
# frozen_string_literal: true

source 'https://rubygems.org'

group :tests, optional: true do
  gem 'activesupport'
  gem 'addressable'
  gem 'json_schemer'
  gem 'nokogiri'
  gem 'rubocop'
  gem 'twitter'
end

gem 'algolia', '~> 2.3'
gem 'dotenv', '~> 2.8'
gem 'parallel', '~> 1.22'
```

## RuboCop version

```SH
$ bundle exec rubocop -V
1.56.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.2) [x86_64-linux-gnu]
```
